cpp/src/arrow/c/abi.h

```diff
@@ -60,6 +60,31 @@ struct ArrowArray {
   void* private_data;
 };
 
+// EXPERIMENTAL
+struct ArrowArrayStream {
+  // Callback to get the stream type
+  // (will be the same for all arrays in the stream).
+  // Return value: 0 if successful, an `errno`-compatible error code otherwise.
+  int (*get_schema)(struct ArrowArrayStream*, struct ArrowSchema* out);
+  // Callback to get the next array
+  // (if no error and the array is released, the stream has ended)
+  // Return value: 0 if successful, an `errno`-compatible error code otherwise.
```

Comment:
silly question is `errno`-compatible a well defined unix/windows term? someplace where I can read more about it?

Reply:
I don't how to phrase it: it returns value that are `errno` error codes (in case of error). A number of values are standard in C++: https://en.cppreference.com/w/cpp/error/errno_macros